psi4/src/psi4/libmints/basisset.cc

```diff
@@ -188,18 +188,20 @@ int BasisSet::n_frozen_core(const std::string &depth, SharedMolecule mol) {
         // will still have 3d electrons active.  Alkali earth atoms will
         // have one valence electron in this scheme.
         for (int A = 0; A < mymol->natom(); A++) {
-            // If this center as an ECP present, move along.
-            if (n_ecp_core(mymol->label(A))) continue;
-            double Z = mymol->Z(A);
-            if (Z > 2) nfzc += 1;
-            if (Z > 10) nfzc += 4;
-            if (Z > 18) nfzc += 4;
-            if (Z > 36) nfzc += 9;
-            if (Z > 54) nfzc += 9;
-            if (Z > 86) nfzc += 16;
-            if (Z > 108) {
+            double Z   = mymol->Z(A);
+            // Add ECPs to Z, the number of electrons less ECP-treated electrons
+            double ECP = n_ecp_core(mymol->label(A));
```

Comment:
So when using ECPs the molecule's function `Z(atom_index)` returns a differnt value? 

That is not something that concerns this PR, but I just find that concerning. Should the molecule care about ECPs? Is that not something for the basis set to care about exclusively?

Reply:
Yup, `Molecule->Z(A)` returns something that is neither the atomic number, nor `0.0` for ghost. That's why my initial pencil-and-paper check of the PR went wrong. I agree disconcerting and worth revisiting. Wish-list maybe?

Reply:
FYI, @bgpeyton, a further variant on `Molecule.Z()`'s behavior.

Reply:
It's probably also the reason why sometimes when using eg. Xe with an ECP, psi4 talks about Fe and sometimes about Xe. I believe the offending part is in Optking:
```
        Writing optimization data to binary file.
        Structure for next step:
        Cartesian Geometry (in Angstrom)
           FE    -1.2297359984  -0.0385946949   0.0000000000
            C     2.5758026217   0.5881468581   0.0000000000
            O     2.2878670730   1.7087928078   0.0000000000
            S     2.9767793599  -0.9094270890   0.0000000000
                        --------------------------
                         OPTKING Finished Execution 
                        --------------------------

    Structure for next step:
    Molecular point group: cs
    Full point group: Cs

    Geometry (in Angstrom), charge = 0, multiplicity = 1:

    XE          -1.232187566437    -0.039755314177     0.000000000000
    C            2.573351053672     0.586986238783     0.000000000000
    O            2.285415504952     1.707632188472     0.000000000000
    S            2.974327791842    -0.910587708272     0.000000000000
```

Reply:
That's funny — we have the only transmuting geometry optimizer. Fortunately, optking's getting replaced this summer.

Reply:
Sorry for my tardiness - I've been traveling nonstop for a couple of weeks.  At the time I threw the ECP stuff together my thinking was that any center with an ECP should have no valence electrons frozen at all.  I think that's what this line is supposed to achieve.  Your fix is much easier to read and allows for some valence electrons to be frozen in addition to the ECP (something that I didn't account for in my haste).  LGTM - thanks for catching this.